src/wmtk/operations/polygon_mesh/Splice.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include <wmtk/PolygonMesh.hpp>
+#include <wmtk/operations/TupleOperation.hpp>
+#include "AtomicOperation.hpp"
+#include "PolygonMeshOperation.hpp"
+
+namespace wmtk::operations {
+namespace polygon_mesh {
+class Splice;
+}
+
+template <>
+struct OperationSettings<polygon_mesh::Splice>
+{
+    // TODO Determine if operation settings are necessary
+};
+
+namespace polygon_mesh {
+
+/**
+ * @class Atomic operation for changing connectivity without changing the number of edges.
+ */
+class Splice : public AtomicOperation
+{
+public:
+    // constructor for default factory pattern construction
+    Splice(Mesh& m, const Tuple& h, const Tuple& g, const OperationSettings<Splice>& settings);
+
+    std::string name() const override;
+
+    using PolygonMeshOperation::hash_accessor;
+
+    /**
+     * @brief Check the precondition that the mesh remains manifold after the splice.
+     *
+     * Splice always produces a manifold connectivity if there are no hole faces, but nonmanifold
+     * vertices are possible on a mesh with boundary.
+     *
+     * @return true if the mesh remains manifold
+     * @return false otherwise
+     */
+    bool precondition();
+
+    /**
+     * @brief Determine if the splice preserves the topology of the original mesh
+     *
+     * @return true if the splice will preserve the topology
+     * @return false otherwise
+     */
+    bool is_topology_preserving();
+
+protected:
+    bool execute() override;
+
+private:
+    const OperationSettings<Splice>& m_settings;
+
+    Tuple m_first_tuple;
+    Tuple m_second_tuple;
+
+    long next_after_splice(long h);
```

Comment:
could you comment what this does a bit more? i didn't undersatnd the function

Reply:
Can do. It's mostly a utility for the precondition; it returns what the next halfedge of `h` would be after the splice without actually touching the `next` attribute yet. It may eventually make sense to replace the precondition with a manifold vertex  after invariant instead to avoid this sort of "next halfedge after the operation" helper function.